src/atomWithMutation.ts

```diff
@@ -0,0 +1,56 @@
+import { DocumentInput } from '@urql/core'
+import type { AnyVariables, Client, OperationContext } from '@urql/core'
+import { atom } from 'jotai/vanilla'
+import type { Getter, WritableAtom } from 'jotai/vanilla'
+import { pipe, subscribe } from 'wonka'
+import { clientAtom } from './clientAtom'
+import {
+  type InitialOperationResult,
+  urqlReactCompatibleInitialState,
+} from './common'
+
+export type AtomWithMutation<
+  Data,
+  Variables extends AnyVariables
+> = WritableAtom<
+  InitialOperationResult<Data, Variables>,
+  [Variables, Partial<OperationContext>] | [Variables],
+  Promise<InitialOperationResult<Data, Variables>>
+>
+
+export function atomWithMutation<
+  Data = unknown,
+  Variables extends AnyVariables = AnyVariables
+>(
+  query: DocumentInput<Data, Variables>,
+  getClient: (get: Getter) => Client = (get) => get(clientAtom)
+): AtomWithMutation<Data, Variables> {
+  const atomDataBase = atom<InitialOperationResult<Data, Variables>>(
+    urqlReactCompatibleInitialState
+  )
+  atomDataBase.onMount = (setAtom) => {
+    return () => {
+      setAtom(urqlReactCompatibleInitialState)
+    }
+  }
```

Comment:
This looks a little weird.
onMount should be paired with onUnmount. (Like React's useEffect.)
Otherwise, this could be a source of bugs.

Reply:
It cleans up cache (only used for paused case) when user redirects from the page. Similarly to react-urql.

Unless I miss on how onMount works, this is the only way I figured how to replicate that behavior.

Reply:
What would it be if we were to setAtom on mount?

Reply:
Same thing as what it is in the cleanup callback :) We just reset it as if it was never in use.

Reply:
Renamed